apis/projectcontour/v1alpha1/contourconfig.go

```diff
@@ -659,6 +662,53 @@ type RateLimitServiceConfig struct {
 	EnableResourceExhaustedCode *bool `json:"enableResourceExhaustedCode,omitempty"`
 }
 
+// TracingConfig defines properties for exporting trace data to OpenTelemetry.
+type TracingConfig struct {
```

Comment:
Please add some validation logic for this struct to `contourconfig_helpers.go` in this same package. In particular, things I see that should be validated:
- OverallSampling must be convertible to a float
- each CustomTag must have exactly one of Literal or RequestHeaderName

(This can look similar to the validation you have in pkg/config).